aicsimageio/transforms.py

```diff
@@ -105,59 +55,116 @@ def reshape_data(
         would have order "STCZYX"
 
     """
-    # Get operator
-    if isinstance(data, da.core.Array):
-        operator = da
-    else:
-        operator = np
+    # Check for parameter conflicts
+    for dim in Dimensions.DefaultOrderList:
+        # return_dims='TCZYX' and fixed dimensions 'C=1'
+        # Dimension is in kwargs
+        # Dimension is an integer
+        # Dimension is in return dimensions
+        if isinstance(kwargs.get(dim), int) and dim in return_dims:
+            raise ConflictingArgumentsError(
+                f"Argument return_dims={return_dims} and "
+                f"argument {dim}={kwargs.get(dim)} conflict. "
+                f"Check usage."
+            )
 
-    # Check for conflicts like return_dims='TCZYX' and fixed channels 'C=1'
-    for dim in return_dims:
-        if kwargs.get(dim) is not None:
+        # return_dims='CZYX' and iterable dimensions 'T=range(10)'
+        # Dimension is in kwargs
+        # Dimension is an iterable
+        # Dimension is not in return dimensions
+        if (
+            isinstance(kwargs.get(dim), (list, tuple, range, slice))
+            and dim not in return_dims
+        ):
             raise ConflictingArgumentsError(
                 f"Argument return_dims={return_dims} and "
                 f"argument {dim}={kwargs.get(dim)} conflict. "
                 f"Check usage."
             )
 
-    # Add each dimension not included in original data
+    # Process each dimension available
     new_dims = given_dims
-    excluded_dims = "".join(set(return_dims) - set(given_dims))
-    for dim in excluded_dims:
-        # Dask doesn't have an explicit expand dims so
-        # we simply reshape with an extra dim
-        data = operator.reshape(data, (1, *data.shape))
-        new_dims = dim + new_dims  # add the missing Dimension to the front
-
-    # If given dims contains a Dimension not in return dims and its depth is 1 remove it
-    # If it's larger than 1 give a warning and
-    # suggest interfacing with the Reader object
-    extra_dims = "".join(set(given_dims) - set(return_dims))
-    for dim in extra_dims:
-        index = new_dims.find(dim)
-        if data.shape[index] > 1:
-            index_depth = kwargs.get(dim)
-            if index_depth is None:
+    getitem_ops = []
+    for dim in given_dims:
+        # Store index of the dim as it is in given data
+        dim_index = given_dims.index(dim)
+
+        # Handle dim in return dims which means that it is
+        # an iterable or None selection
+        if dim in return_dims:
+            # Specific iterable requested
+            if dim in kwargs:
+                # Actual dim operator
```

Comment:
what does operator mean?  Is that a standard term for the union of int, list, tuple, slice, and range?

Reply:
Renamed and updated comment to `dim_spec`